packages/text-annotator-react/src/TextAnnotator.tsx

```diff
@@ -26,7 +29,9 @@ export const TextAnnotator = <E extends unknown>(props: TextAnnotatorProps<E>) =
   useEffect(() => {
     if (!setAnno) return;
 
-    const anno = createTextAnnotator(el.current, opts);
+    const adapter = typeof opts.adapter === 'function' ? opts.adapter(el.current) : opts.adapter;
```

Comment:
I'm not using a more laconic `opts` overriding
```ts
opts.adapter = typeof opts.adapter === 'function' ? opts.adapter(el.current) : opts.adapter;
```
because it would mutate the props